src/main/scala/lms/thirdparty/cuda/cuda.scala

```diff
@@ -812,6 +813,28 @@ trait CudaOps extends Dsl with StackArrayOps with SizeTOps with CLibs with CudaF
       }
     }
 
+  def cudaTranspose[N:Numeric:Manifest](implicit __pos: SourceContext) = cudaGlobalFun {
+    (in: Rep[Array[N]], out: Rep[Array[N]]) => 
```

Comment:
Nit: we would want some comments here via `generate_comment(xxx)` function. Those comments will be generated in the generated code, explaining what the parameters are and how the caller is supposed to call it (what is the Dim3 used for instance)